cpp/src/parquet/arrow/writer.h

```diff
@@ -114,6 +115,14 @@ class PARQUET_EXPORT FileWriter {
   virtual ::arrow::Status WriteColumnChunk(
       const std::shared_ptr<::arrow::ChunkedArray>& data) = 0;
 
+  /// \brief Start a new buffered row group.
+  ///
+  /// Returns an error if not all columns have been written.
+  virtual ::arrow::Status NewBufferedRowGroup() = 0;
+
+  /// \brief Write a RecordBatch into the buffered row group.
```

Comment:
Should clarify:

 * That multiple batches can be written to the same row group.
 * What happens when you write more data than `WriterProperties.max_row_group_length`?
 * When do the batches get flushed to the output stream?

```suggestion
  /// \brief Write a RecordBatch into the buffered row group.
  ///
  /// Multiple RecordBatches can be written into the same row
  /// group through this method. 
```

Reply:
Added more comments to clarify all behaviors mentioned above.